thingiverse/ThingiverseService.py

```diff
@@ -41,31 +48,45 @@ class ThingiverseService(QObject):
     # Signal triggered when a file has started or stopped downloading.
     downloadingStateChanged = pyqtSignal()
 
+    # Signal triggered when the API Client is switched
+    apiClientChanged = pyqtSignal()
+
     def __init__(self, extension: "ThingiverseExtension", parent=None):
         super().__init__(parent)
         self._extension = extension  # type: ThingiverseExtension
 
         # Hold the things found in query results.
-        self._things = []  # type: List[JSONObject]
+        self._things = []  # type: List[Thing]
         self._query = ""  # type: str
         self._query_page = 1  # type: int
         self._is_from_collection = False  # type: bool
 
         # Hold the thing and thing files that we currently see the details of.
-        self._thing_details = None  # type: Optional[JSONObject]
-        self._thing_files = []  # type: List[JSONObject]
+        self._thing_details = None  # type: Optional[Thing]
+        self._thing_files = []  # type: List[ThingFile]
         self._is_downloading = False  # type: bool
-
+        
         # The API client that we do all calls to Thingiverse with.
-        self._api_client = ThingiverseApiClient()  # type: ThingiverseApiClient
+        self._api_client = ThingiverseApiClient()  # type: ApiClient
 
-        # Register plugins settings.
-        CuraApplication.getInstance().getPreferences().preferenceChanged.connect(self._onPreferencesChanged)
-        self._user_name = self._initSettings(Settings.SETTINGS_USER_NAME_PREFERENCES_KEY)
+        self._initSettings(Settings.THINGIVERSE_USER_NAME_PREFERENCES_KEY, "")
+        self._initSettings(Settings.MYMINIFACTORY_USER_NAME_PREFERENCES_KEY, "")
 
         # List of supported file types.
         self._supported_file_types = []  # type: List[str]
 
+    @pyqtSlot(str, name="setService")
+    def setService(self, value: str) -> None:
```

Comment:
Instead of this we could keep a dict of 'providers' and call the right function on the right provider depending on the actively selected service.